store/policy.go

```diff
@@ -78,6 +78,75 @@ func (s *Store) GetPolicy(ctx context.Context, find *api.PolicyFind) (*api.Polic
 	return policy, nil
 }
 
+// PatchPolicy patchs an instance of Policy by PolicyPatch.
+func (s *Store) PatchPolicy(ctx context.Context, patch *api.PolicyPatch) (*api.Policy, error) {
+	policyRaw, err := s.patchPolicyRaw(ctx, patch)
+	if err != nil {
+		return nil, fmt.Errorf("failed to patch policy with PolicyPatch[%+v], error[%w]", patch, err)
+	}
+	policy, err := s.composePolicy(ctx, policyRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compose policy with policyRaw[%+v], error[%w]", policyRaw, err)
+	}
+	return policy, nil
+}
+
+// DeletePolicy deletes an existing ARCHIVED policy by PolicyDelete.
+func (s *Store) DeletePolicy(ctx context.Context, delete *api.PolicyDelete) error {
+	// Validate policy.
+	// Currently we only support PolicyTypeSchemaReview type policy to delete by id
+	if delete.Type != api.PolicyTypeSchemaReview {
```

Comment:
Also need to check we only allow delete if row status is ARCHIVED

Reply:
The `row_status` limit is put into the `deletePolicyImpl`. Do I still need to get the policy and check the `row_status` here? (seems it cannot guarantee the row status)

Reply:
I missed that. I would prefer to put the validation logic inside a single method. Seems here is more appropriate since we haven't started the tx.

Reply:
Updated, add the check for `row_status` before calling the `deletePolicyImpl`, but still keep the `row_status` in the delete query.